### PR TITLE
LLVM17LIT

### PR DESCRIPTION
**Pull request by nishant-sachdeva**
- Adding LIT based integration testing to replace `make verify-all` based testing

### CODE REVIEW

File: docs/spec_compilation.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# README.md
+
+- Here, we describe the process to follow to generate ll files from the `cpu-spec-2006` and the `cpu-spec-2017` folders.
+
+## CPU 2006
+
+- Run installation using the command `bash install.sh`.
+- Create config file, using `config_clang14_cpu2006.cfg`. Change compilation parameters to appropriate llvm version.
```
Comment (svkeerthy) [inline docs/spec_compilation.md:8]: Can you please list the parameters as well..

File: docs/spec_compilation.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- Copy the config file to the `config` folder in the `cpu2006install` folder.
+- Source the install. `source shrc`
+- Run the command
+  - `runspec --config config_clang16_cpu2006.cfg  --tune=base  --action=build --rebuild --define build_ncpus=1 int fp`
+- Once the `.o` files are generated, run the script (`get_ll_spec.sh`) with appropriate changes to get these files together in the repo.
```
Comment (svkeerthy) [inline docs/spec_compilation.md:13]: where is get_ll_spec.sh? Please add the link.

File: docs/spec_compilation.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## CPU 2017
+
+- Run installation.
+- Create config file, using `config_clang14_cpu2017.cfg`. Change compilation parameters to appropriate llvm version.
```
Comment (svkeerthy) [inline docs/spec_compilation.md:18]: Similar changes here please.

File: docs/version_upgrade_process.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+The following guide details the steps followed in the upgrade process for IR2Vec.
```
Comment (svkeerthy) [inline docs/version_upgrade_process.md:1]: "The following guide details the steps followed in upgrading the LLVM version supported by IR2Vec"

File: docs/version_upgrade_process.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+# Compile the IR2Vec binary
+   - Instructions for the same are present [here](https://github.com/IITH-Compilers/IR2Vec/blob/main/README.md)
+
+# Generating .ll files for re-training.
+   - The git repo `IR2Vec-Version-Upgrade-Checks` has the required scripts to be run for this process.
+   - The repo is available [here](https://github.com/IITH-Compilers/IR2Vec-Version-Upgrade-Checks/)
+   - Our relevant scripts and files will be present in the folder `collect_ir`.
```
Comment (svkeerthy) [inline docs/version_upgrade_process.md:9]: Please make it clear.

File: docs/version_upgrade_process.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+   - We use the C++ Library Boost, and the CPU-SPEC source codes to generate training data.
+      - Download these source codes.
+   - Compile the relevant Boost `.c*` files with the relevant LLVM version.
+      - The folder has the script `get_boost_ll.py` for this purpose.
+   - CPU/SPEC .c++ files compilation with the relevant LLVM version.
+      - For detailed instructions on this step, refer to [here](https://github.com/nishant-sachdeva/IR2Vec/wiki/spec_compilation).
```
Comment (svkeerthy) [inline docs/version_upgrade_process.md:15]: Please fix the link to point to IITH-Compilers/IR2Vec

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: Manylinux2014_Compliant_Source/pkg/build.sh [hunk 7]
```diff
@@ -20,5 +20,8 @@ cmake -DCMAKE_BUILD_TYPE=Release .. && make -j"$(nproc)" && make install
 cd ..
 cp build/vocabulary.h Manylinux2014_Compliant_Source/pkg/ir2vec/
 cp src/include/utils.h Manylinux2014_Compliant_Source/pkg/ir2vec/
+cp src/include/IR2Vec.h Manylinux2014_Compliant_Source/pkg/ir2vec/
 cp build/src/version.h Manylinux2014_Compliant_Source/pkg/ir2vec/
+cp vocabulary/seedEmbeddingVocab.txt Manylinux2014_Compliant_Source/pkg/ir2vec/
```
Comment (svkeerthy) [inline Manylinux2014_Compliant_Source/pkg/build.sh:25]: This vocabulary need not be copied anymore. Please check and remove.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: Manylinux2014_Compliant_Source/pkg/build.sh [hunk 8]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+s#!/bin/bash
```
Comment (svkeerthy) [inline Manylinux2014_Compliant_Source/pkg/build.sh:1]: Please fix this.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: Manylinux2014_Compliant_Source/pkg/build.sh (hunk 8 above)
Comment (nishant-sachdeva) [reply Manylinux2014_Compliant_Source/pkg/build.sh]: Pushed changes for this

**Pull request merged**